plugins/catalog-backend/src/ingestion/processors/util/github.ts

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GroupEntity, UserEntity } from '@backstage/catalog-model';
+import { graphql } from '@octokit/graphql';
+
+// Graphql types
+
+type QueryResponse = {
+  organization: Organization;
+};
+
+type Organization = {
+  membersWithRole: Connection<User>;
+  team: Team;
+  teams: Connection<Team>;
+};
+
+type PageInfo = {
+  hasNextPage: boolean;
+  endCursor?: string;
+};
+
+type User = {
+  login: string;
+  bio?: string;
+  avatarUrl?: string;
+  email?: string;
+  name?: string;
+};
+
+type Team = {
+  slug: string;
+  combinedSlug: string;
+  description?: string;
+  parentTeam?: Team;
+  members: Connection<User>;
+};
+
+type Connection<T> = {
+  pageInfo: PageInfo;
+  nodes: T[];
+};
+
+/**
+ * Gets all the users out of a GitHub organization.
+ *
+ * Note that the users will not have their memberships filled in.
+ *
+ * @param client An octokit graphql client
+ * @param org The slug of the org to read
+ */
+export async function getOrganizationUsers(
+  client: typeof graphql,
+  org: string,
+): Promise<{ users: UserEntity[] }> {
+  const users: UserEntity[] = [];
+
+  // There is no user -> teams edge, so we leave the memberships empty for
+  // now and let the team iteration handle it instead
+  let cursor: string | undefined = undefined;
+  const query = `
+      query users($org: String!, $cursor: String) {
+        organization(login: $org) {
+          membersWithRole(first: 100, after: $cursor) {
+            pageInfo { hasNextPage, endCursor }
+            nodes { avatarUrl, bio, email, login, name }
+          }
+        }
+      }`;
+
+  for (let i = 0; i < 100 /* just for sanity */; ++i) {
+    const response: QueryResponse = await client(query, {
+      org,
+      cursor,
+    });
+
+    const connection = response.organization?.membersWithRole;
+    if (!connection) {
+      throw new Error(`Found no organization named ${org}`);
+    }
+
+    for (const user of connection.nodes) {
+      const entity: UserEntity = {
+        apiVersion: 'backstage.io/v1alpha1',
+        kind: 'User',
+        metadata: {
+          name: user.login,
+          annotations: {
+            'github.com/user-login': user.login,
```

Comment:
Same as for teams, thoughts on `github.com/login`?

Reply:
Well then as you see you would still have two different ones. Could one instead then have `backstage.io/github-identifier` for both? Or `github.com/identity-match` or whatever?

I'm not sure where I stand on this. Just `github.com/slug` felt weird. It's nice to see an annotation and knowing from its name what it really stands for. So reading it out, the original ones felt the most natural and understandable.

Reply:
Yeah got no problem with keeping `team-slug`, the thought of whether annotations should be documented globally or per kind just popped up in my head.

One thing to consider is that iirc, even github orgs have a `login`, so we might end up with both `user-login` and `org-login`. I think I prefer just `login` in that case. Should definitely keep the "login" word there though, as that's what GitHub uses.

Reply:
oh didn't change this one btw for now, let's discuss separately after this pr what's the best pattern